src/Vault.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "lib/chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "lib/yield-utils-v2/contracts/token/IERC20.sol";
+import "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+
+contract Vault is Ownable {
+
+    ///@dev Vault records collateral deposits of each user
+    mapping (address => uint) public deposits; 
+
+    ///@dev Vault records debt holdings of each user
+    mapping (address => uint) public debts;  
+
+
+    ///@notice ERC20 interface specifying token contract functions
+    ///@dev For constant variables, the value has to be fixed at compile-time, while for immutable, it can still be assigned at construction time.
```

Comment:
i know its tough to think of @dev notes but u dont need to provide basic info about common syntax lol

Reply:
lol I kept that around for myself tbh, so it would stick in my brain as a habit. will remove ;)

Reply:
fair enough!